dGame/dComponents/TriggerComponent.cpp

```diff
@@ -0,0 +1,214 @@
+#include "TriggerComponent.h"
+
+#include "dZoneManager.h"
+#include "LUTriggers.h"
+#include "eTriggerCommandType.h"
+
+#include "MissionComponent.h"
+#include "PhantomPhysicsComponent.h"
+
+#include "CDMissionTasksTable.h"
+
+
+TriggerComponent::TriggerComponent(Entity* parent, const std::string triggerInfo): Component(parent) {
+	m_Parent = parent;
+	m_Trigger = nullptr;
+
+	std::stringstream ss(triggerInfo);
+	std::vector<std::string> tokens;
+	std::string token;
+	while (std::getline(ss, token, ':')) {
+		tokens.push_back(token);
+	}
+
+	uint32_t sceneID;
+	GeneralUtils::TryParse<uint32_t>(tokens[0], sceneID);
+	uint32_t triggerID;
+	GeneralUtils::TryParse<uint32_t>(tokens[1], triggerID);
+
+	if (m_Trigger) {
+		delete m_Trigger;
+		m_Trigger = nullptr;
+	}
```

Comment:
https://github.com/DarkflameUniverse/DarkflameServer/pull/986/files#diff-c6147890efa9c480edd892491e910cc7eb185bb3675936183bb28999b3f4e2c8R15
Not sure what this is supposed to be doing - in a constructor, this will never be null as per the above line so this code block will never fire off

Reply:
glad I'm not the only one who thought this was weird, I'm going to remove it